From: Jason Rhinelander <jason@imaginary.ca>
Date: Tue, 29 Sep 2020 17:51:40 -0300
Subject: boost clang build hack

---
 cmake/StaticBoost.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/StaticBoost.cmake b/cmake/StaticBoost.cmake
index 21407a3..b56b284 100644
--- a/cmake/StaticBoost.cmake
+++ b/cmake/StaticBoost.cmake
@@ -120,7 +120,7 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
 else()
   message(FATAL_ERROR "don't know how to build boost with ${CMAKE_CXX_COMPILER_ID}")
 endif()
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/user-config.bjam "using ${boost_toolset} : : ${deps_cxx} ;")
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/user-config.bjam "using ${boost_toolset} : : ${deps_cxx} : <ranlib>llvm-ranlib-8 <archiver>llvm-ar-8 ;")
 
 set(boost_patch_commands "")
 if(APPLE AND BOOST_VERSION VERSION_LESS 1.74.0)
